napi-inl.h

```diff
@@ -2242,7 +2242,16 @@ inline PropertyDescriptor::operator const napi_property_descriptor&() const {
 ////////////////////////////////////////////////////////////////////////////////
 
 template <typename T>
-inline ObjectWrap<T>::ObjectWrap() {
+inline ObjectWrap<T>::ObjectWrap(Napi::CallbackInfo callbackInfo) {
+  napi_env env = callbackInfo.Env();
+  napi_value wrapper = callbackInfo.This();
+  napi_status status;
+  napi_ref ref;
+  status = napi_wrap(env, wrapper, this, FinalizeCallback, nullptr, &ref);
```

Comment:
`static_cast<>` can _change the pointer_ for objects that have a vtable. Because `ObjectWrap::Unwrap()` interprets the `void*` as a `T*`, we should be sure to store it as a `T*`.